dts/bindings/misc/nxp,inputmux.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+include: [base.yaml]
+
+compatible: "nxp,inputmux"
+description: NXP i.MX INPUTMUX driver. Routes internal SoC signals.
+
+properties:
+  reg:
+    required: true
+
+child-binding:
```

Comment:
these should not be child bindings, Linux guidelines (which Zephyr project tries to follow) strongly discourages making nodes just to initialize drivers, each node should represent a device, not a bit in a register